### PR TITLE
Revert the status.outputs key back to object from array

### PR DESCRIPTION
**Pull request by pvdvreede**
Make sure status.output in crd update status is an object. 
Outputs provides key value pairs as information to a user to be able to
pull out. 

This PR reverts its type back to an object instead of an
array as each item in the outputs key should be accessible
programmatically for a delivery team developer to pull out if required,
so an array would make this hard as ordering might not be guaranteed.

### CODE REVIEW

File: test/tests/test-upsert [hunk 1]
```diff
@@ -4,11 +4,12 @@ set -eo pipefail
 
 kubectl apply -f /e2e/mocks/exampleresource.yml
 
-sleep 5
+sleep 8
```
Comment (colinliddlemyob) [inline test/tests/test-upsert]: Why was this changed?

Comment (pvdvreede) [reply test/tests/test-upsert]: Good question.

This is a hackety hack as I noticed the test was a bit flakey.

This should really be fixed to properly wait for the object to be updated with a timeout - but in a separate PR.

So for the moment, I increased the hackiness.
